tools/odrive/utils.py

```diff
@@ -118,11 +118,106 @@ def did_close(evt):
     plot_t = threading.Thread(target=plot_data)
     plot_t.daemon = True
     plot_t.start()
-    
 
     return cancellation_token;
     #plot_data()
 
+
+class BulkCapture:
+    '''
+    Asynchronously captures a bulk set of data when instance is created.
+
+    get_var_callback: a function that returns the data you want to collect (see the example below)
+    data_rate: Rate in hz
+    length: Length of time to capture in seconds
```

Comment:
I'd call this `duration` to distinguish it more clearly from something like "buffer length".